### PR TITLE
Feature/3423 app bar example page

### PR DESCRIPTION
**Pull request by bkarambe**
Fixes # BLUI-3423

#### Changes proposed in this Pull Request:
- Added example page for App Bar

#### Screenshots:
Screenshot 2022-09-19 at 7 31 18 PM

#### To Test:
- yarn start:reactdev

### CODE REVIEW

**Comment by emclaug2** [comment]
Here's a couple things I've noticed:

1. Can you remove the horizontal scrollbar that is appearing on windows?

image

2.  "A <AppBar> with a collapsed variant..." should be  "**An** <AppBar> with a collapsed variant..."

3. On the collapsed variant, the way the example is resized differs from angular's example.  The entire example window shrinks rather than just the appbar portion.

4.  No custom content example? 
https://blui-angular-docs--pr506-feature-3439-app-bar-6a6cbguv.web.app/components/app-bar/examples

**Comment by bkarambe** [comment]
> Here's a couple things I've noticed:
> 
> 1. Can you remove the horizontal scrollbar that is appearing on windows?
> 
> image
> 
> 2. "A AppBar with a collapsed variant..." should be  "**An** AppBar  with a collapsed variant..."
> 3. On the collapsed variant, the way the example is resized differs from angular's example.  The entire example window shrinks rather than just the appbar portion.
> 4. No custom content example?
>    https://blui-angular-docs--pr506-feature-3439-app-bar-6a6cbguv.web.app/components/app-bar/examples

I did not add Custom Content example, because App Bar no do accept props like icon. Can we add a Complex App Bar instead which should have children like Icon, Threeliner and Actions?

**Comment by huayunh** [comment]
> Can we add a Complex App Bar instead which should have children like Icon, Threeliner and Actions?

Yes, I would say so. Try to mirror Angular for this example.

File: docs/src/componentDocs/AppBar/examples/CustomContentAppBarExample.tsx [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+                    <ThreeLiner
+                        title={'Title'}
+                        subtitle={'Subtitle'}
+                        info={'Info'}
+                        animationDuration={300}
+                        sx={{
+                            top: 0,
+                            position: 'relative',
+                            flexGrow: 1,
+                            marginLeft: 7,
```
Comment (huayunh) [inline docs/src/componentDocs/AppBar/examples/CustomContentAppBarExample.tsx:58]:
```suggestion
                            top: 0,
                            position: 'relative',
                            marginLeft: 4,
```

File: docs/src/componentDocs/AppBar/examples/CustomContentAppBarExample.tsx [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+                        </IconButton>
+                    </Box>
+                </Toolbar>
+            </AppBar>
+            <Box
+                id="appbarBodyFiller2"
+                sx={{
+                    height: 400,
+                    overflow: 'scroll',
+                    backgroundColor: (theme) => theme.palette.background.paper,
```
Comment (huayunh) [inline docs/src/componentDocs/AppBar/examples/CustomContentAppBarExample.tsx:80]: MUI v5 should allow you to do 

```
backgroundColor: 'background.paper',
```

here, if you are looking to simplify the code

File: docs/src/componentDocs/AppBar/examples/CustomContentAppBarExample.tsx [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+                    />
+                    <Spacer />
+                    <Box style={{ display: 'flex', flexDirection: 'row' }}>
+                        <IconButton color={'inherit'}>
+                            <Search />
+                        </IconButton>
+                        <IconButton color={'inherit'}>
+                            <Download />
+                        </IconButton>
+                        <IconButton color={'inherit'}>
```
Comment (huayunh) [inline docs/src/componentDocs/AppBar/examples/CustomContentAppBarExample.tsx:69]:
```suggestion
                        <IconButton color={'inherit'}  edge={'end'}>
```

File: docs/src/componentDocs/AppBar/examples/CustomContentAppBarExample.tsx [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+                    },
+                    '&.expanded': {
+                        '& .BluiThreeLiner-root': {
+                            top: 64,
+                        },
+                    },
+                    width: 450,
+                    mx: 'auto',
+                    zIndex: 'auto',
+                }}
```
Comment (huayunh) [inline docs/src/componentDocs/AppBar/examples/CustomContentAppBarExample.tsx:40]: I am noticing a lot of these `sx` styles are missing from the code snippet. You either show them all, or you hide everything and tell people to "view full code on GitHub" (which I'd be inclined to do). Showing half of the available styles can be misleading to our users.

**Review by huayunh — APPROVED** [review]

**Pull request merged**